qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/customers/CustomerDbConnector.groovy

```diff
@@ -81,23 +152,122 @@ class CustomerDbConnector implements CreateCustomerDataSource, UpdateCustomerDat
+    String query = "SELECT * FROM customer WHERE first_name = ? AND last_name = ? AND email = ?"
+    Connection connection = connectionProvider.connect()
+
+    def customerAlreadyInDb = false
+
+    connection.withCloseable {
+      def statement = connection.prepareStatement(query)
+      statement.setString(1, customer.firstName)
+      statement.setString(2, customer.lastName)
+      statement.setString(3, customer.eMailAddress)
+      statement.execute()
+      def result = statement.getResultSet()
+      customerAlreadyInDb = result.next()
+    }
+    return customerAlreadyInDb
+  }
+  private static int createNewCustomer(Connection connection, Customer customer) {
+    String query = "INSERT INTO customer (first_name, last_name, title, email) " +
+            "VALUES(?, ?, ?, ?)"
+
+    List<Integer> generatedKeys = []
+
+    def statement = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)
+    statement.setString(1, customer.firstName )
+    statement.setString(2, customer.lastName)
+    statement.setString(3, customer.title.value)
+    statement.setString(4, customer.eMailAddress )
+    statement.execute()
+    def keys = statement.getGeneratedKeys()
+    while (keys.next()){
+      generatedKeys.add(keys.getInt(1))
+    }
+
+    return generatedKeys[0]
+  }
+  private static void storeAffiliation(Connection connection, int customerId, List<Affiliation>
+          affiliations) {
+    String query = "INSERT INTO customer_affiliation (affiliation_id, customer_id) " +
+            "VALUES(?, ?)"
+
+    affiliations.each {affiliation ->
+      def affiliationId = getAffiliationId(connection, affiliation)
+      def statement = connection.prepareStatement(query)
+      statement.setInt(1, affiliationId)
+      statement.setInt(2, customerId)
+      statement.execute()
+
+    }
+  }
+  //Fixme no use of closeable
+  private static int getAffiliationId(Connection connection, Affiliation affiliation) {
+    String query = "SELECT * FROM affiliation WHERE organization=? " +
+            "AND address_addition=? " +
+            "AND street=? " +
+            "AND postal_code=? " +
+            "AND city=?"
+
+    List<Integer> affiliationIds = []
+
+    def statement = connection.prepareStatement(query)
+    statement.setString(1, affiliation.organisation)
+    statement.setString(2, affiliation.addressAddition)
+    statement.setString(3, affiliation.street)
+    statement.setString(4, affiliation.postalCode)
+    statement.setString(5, affiliation.city)
+    statement.execute()
+    ResultSet rs = statement.getResultSet()
+    while (rs.next()) {
+      affiliationIds.add(rs.getInt(1))
+    }
+
+    if(affiliationIds.size() > 1) {
+      throw new DatabaseQueryException("More than one entry found for $affiliation.")
+    }
+    if (affiliationIds.empty) {
+      throw new DatabaseQueryException("No matching affiliation found for $affiliation.")
+    }
+    return affiliationIds[0]
```

Comment:
I would appreciate, if the try-with-ressource around the connection is used here.

Reply:
I will not address this in this PR since this PR only moves existing code into the DBConnector.

Reply:
Issue?